scipy/optimize/lbfgsb.py

```diff
@@ -359,6 +359,13 @@ def func_and_grad(x):
         else:
             break
 
+    # f should really always be a scalar
+    f = np.asarray(f)
+    if len(f.shape) > 1:
+        warnings.warn("The LBFGS-B fun value used to be a 1x1 array. It is now a scalar.",
```

Comment:
Was it shape `(1,1)` or just `(1,)` before?

Reply:
`(1,)` in a test case. I've fixed the test now and adapted the warning message.